### ISSUE TITLE
Bring back Clarity CodeCov Table on PR Conversations

### ISSUE THREAD

**Issue by jo-tm** [Trust-Machines/stacks-sbtc#611]
**Is your feature request related to a problem? Please describe.**

Using Github Action we use to have a nice CodeCov tabled displayed on PRs conversations, not is not working. Bring it back!

We use to have something like this goodie:
![image](https://github.com/Trust-Machines/stacks-sbtc/assets/30682875/2ad05994-879f-4af2-bbc6-32f3928531e9)

**Describe the solution you'd like**

Review what happended with the GH Action and fix it.

**Describe alternatives you've considered**

There are many version of CodeCov GH module...

**Additional context**

There is a text only table but hidden in the actions outputs, inconvenient to find and look...

**Comment by friedger** [comment]
For clarity code there is a script `coverage-report.sh` to generate coverage data

**Comment by jo-tm** [comment]
yep, I think is also useful to see the coverage Delta on the PRs.

**Comment by jo-tm** [comment]
@AshtonStephens wants to track this effort.

**Comment by jo-tm** [comment]
@AshtonStephens any idea on how to fix this? I was using `https://github.com/Trust-Machines/stacks-sbtc/blob/main/.github/workflows/clarinet.yml` but something changed on another file and break it.

**Comment by AshtonStephens** [comment]
@friedger I've been looking at the clarity workflow for a bit and would prefer the workflow utilize the canonical method of adding clarinet testing in a CI (below) as opposed to what it does now.

```yaml
name: CI
on: [push]
jobs:
  tests:
    name: "Test contracts with Clarinet"
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: "Execute unit tests"
        uses: docker://hirosystems/clarinet:latest # <----- This bit
        with:
          args: test --coverage --manifest-path=./Clarinet.toml
      - name: "Export code coverage"
        uses: codecov/codecov-action@v1
        with:
          files: ./coverage.lcov
          verbose: true
```

It looks like you recently worked on a number of scripts surrounding the clarinet testing for sbtc-mini. Do you think getting codecov and clarinet testing within our CI something you could give a quick look?

**Comment by AshtonStephens** [comment]
@friedger I had a conversation with @jo-tm and I understand more what's going on now with the scripts and how we're writing our tests in clarity - that makes sense! I'll put out a PR that uploads the code to codecov.

**Comment by AshtonStephens** [comment]
Somehow these stopped showing up between https://github.com/Trust-Machines/stacks-sbtc/pull/324 and https://github.com/Trust-Machines/stacks-sbtc/pull/376. I'm not seeing anything remotely related to codecov that changed between these two PRs.

**Comment by AshtonStephens** [comment]
**TL;DR:** I have to conclude that it's an issue with codecov's GitHub integration for the entire Trust-Machines organization and will require Codecov support to fix. We have no configuration that justifies us losing a the PR message between the two PRs and are observing the same issue across repositories (https://github.com/Trust-Machines/wsts/pull/17).

----

Someone found a drastic solution to the issue a while ago [codecov-not-commenting-in-prs](https://community.codecov.com/t/codecov-not-commenting-in-prs/3496). It's very aggressive and requires deleting repos and disabling codecov for the entire organization.

To summarize, he:
1. Disabled codecov for every one of his org's repos
2. Deleted the old repos and made new ones.

I recommend against this for us, especially if it means we might lose the history of github PRs for our repos.

A few things I tried that make me nearly certain that this is an issue with the entirety of `Trust-Machines` gh org and not with this specific repo:
1. I tried seeing if codecov would work if we transferred our code to a new repo within Trust Machines and had no luck: https://github.com/Trust-Machines/github-actions-playground/pull/1.
2. I moved the sbtc repo on the stacks-repo which has working codecov comments ([example](https://github.com/stacks-network/sbtc/pull/44)) into a repo on the Trust-Machines github org. No codecov comments: https://github.com/Trust-Machines/github-actions-playground/pull/2

It seems that this is an issue in all gh repos under the TM organization. For example, this PR https://github.com/Trust-Machines/wsts/pull/17 on the wsts repo should have a codecov comment ([codecov upload in CI](https://github.com/Trust-Machines/wsts/blob/main/.github/workflows/ci.yml#L46-L48)) but doesn't.

(Oddly [this PR](https://github.com/Trust-Machines/wsts/pull/18) in the wsts repo **does** have a codecov comment, and is well after the first time we lose codecov PR comments. I've got no justification for that appearing and other codecov comments not.)

***Since this issue will require more work than expected I'm going to deprioritize it for now and ensure that we get back to it when we have less pressing deadlines or we decide to re-prioritize it.***
